### ISSUE TITLE
[suggestion] One-way glass that can camo sides to air.

### ISSUE THREAD

**Issue by fuzzyweapon** [Wyn-Price/SecretRooms#16]
This would work similar to one-way glass but just be able to be completely invisible on the camo-ed/activated sides instead of forcing it to take on the visual attributes of the nearest solid block.

As a sidenote...
I was able to place a barrier block, then one way glass on top of this and the one-way glass block immediately took on the visual properties of the barrier block in all directions, but was completely see through.  Activating a side did not cause any change, so this is obviously a bug, but probably one not worth fixing...just, sorta related tangentially.

**Comment by Wyn-Price** [comment]
I think the barrier not being rendered it due to the barrier marking the block to not be rendered. Essentially, would this new block just be normal one way glass, but without rendering the glass texture?

**Comment by fuzzyweapon** [comment]
Yeah that's what I thought too, but it was fun chaining various camo blocks to the one-way glass that was camoing to the barrier bc it made all of them not render. XD

Hmm, I think the best way to explain this would be a little user scenario.

Imagine you set some one way glass down, then you have the ability to use/right-click a face to make it a solid-camo or shift+right-click/sneak+use to instead make a face look as if nothing is there when the player is looking at it from that angle.  I haven't looked at the source code, so while this might be really cool for the user, it might not be maintainable to shove this functionality into the existing one-way glass.


Some fun ideas this could unlock, though probably the tip of the iceberg...
Allow for floating decorative bits that need a solid block to adhere too (grass is a terrible example but it is what is coming into my head).

Thinking about using this with the redstone camo-visibility idea I submitted, you could literally build structures hidden in the clouds, spooky ruins that only show up when the sun goes down, or set up a dirt hovel with a BUNCH OF DEFENSES and surveillance (a la Security Craft) then when some troll shows up to screw with you again, suddenly drop the facade.

If you had the ability to combine the redstone camo-visibility toggle idea with having two different camo states on the same block....think...lava below, setup some one-way glass to camo stone bricks above it.  Toggle the redstone and while the camo stone is off, set all faces of the bricks to have invisi-camo with shift+right-click so that when the redstone is off it looks like a stone floor, but when the redstone is on, it looks like they are floating in air above the lava.  This implies that each face could carry two separate states for camo that co-exist.  I would think if a player tries to, say, put invisi-camo on top of solid-block/default camo that the invisi-camo would just clear the default and set the invisi for that redstone state.

Anyways, this kind of spells out the kind of weird cool opportunities that it could unlock.

**Comment by fuzzyweapon** [comment]
Obviously some of these more advanced ideas could _only_ exist if the functionality is integrated with the existing one-way glass.
